libqtile/widget/redshift.py

```diff
@@ -0,0 +1,333 @@
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import subprocess
+
+from libqtile.command.base import expose_command
+from libqtile.log_utils import logger
+from libqtile.widget import base
+from libqtile.widget.base import _TextBox
+
+
+class GammaGroup:
+    """
+    GammaGroup is a helper class to group redshift gamma settings.
+    """
+
+    def __init__(self, red, green, blue):
+        self.gamma_red = red
+        self.gamma_green = green
+        self.gamma_blue = blue
+
+    def _format_gamma(self, format_txt: str) -> str:
+        return format_txt.format(
+            gamma_red=self.gamma_red,
+            gamma_green=self.gamma_green,
+            gamma_blue=self.gamma_blue,
+        )
+
+    def _redshift_fmt(self) -> str:
+        return f"{self.gamma_red}:{self.gamma_green}:{self.gamma_blue}"
+
+    def __repr__(self) -> str:
+        return f"Gamma: {self._redshift_fmt()}"
+
+    def __str__(self) -> str:
+        return (
+            f"GammaGroup(red={self.gamma_red}, green={self.gamma_green}, blue={self.gamma_blue})"
+        )
+
+
+class RedshiftDriver:
+    """
+    RedshiftDriver is a helper class
+    that executes all of the interactions with redshift.
+    """
+
+    def enable(self, temperature, gamma: GammaGroup, brightness):
+        """
+        Run redshift command with given parameters
+        """
+        subprocess.run(
+            [
+                "redshift",
+                "-P",
+                "-O",
+                str(temperature),
+                "-b",
+                str(brightness),
+                "-g",
+                gamma._redshift_fmt(),
+            ],
+            check=True,
+        )
+
+    def reset(self):
+        """
+        Call reset on redshift to reset to default settings.
+        """
+        subprocess.run(["redshift", "-x"], check=True)
+
+
+class Redshift(_TextBox, base.PaddingMixin):
+    """
+    Redshift widget provides the following functionality:
+      - Call redshift with a specific brightness, temperature and gamma config without using your location.
+
+    The redshift command can be called by just left-clicking the widget and disabling it the same way.
+
+    Scrolling through the widget will show the settings,
+    ie. the brightness, gamma and temperature that redshift
+    will use if the widget is enabled.
+    """
+
+    redshift_driver = RedshiftDriver()
+    orientations = base.ORIENTATION_HORIZONTAL
```

Comment:
I think this can be removed. Text boxes should display fine both horizontally and vertically.
```suggestion

```